### PR TITLE
LOG-4011: [release-5.7] [Vector] Collector not complying with the custom tlsSecurityProfile configuration

### PR DESCRIPTION
**Pull request by syedriko**
Commented out a few unused VRL functions to avoid dependency on the onig-sys crate which fails to build on Fedora 38

For "min_tls_version" and "ciphersuites" TLS settings
  - added client-side implementation
  - cleaned up the server-side implementation. Now the TLS 1.3 default ciphersuites are left to OpenSSL.

JIRA: https://issues.redhat.com/browse/LOG-4011
Related PR: https://github.com/ViaQ/vector/pull/141

### CODE REVIEW

File: lib/vrl/stdlib/Cargo.toml [hunk 1]
```diff
@@ -124,8 +124,9 @@ default = [
     "parse_csv",
     "parse_duration",
     "parse_glog",
-    "parse_grok",
-    "parse_groks",
+# REVISIT syedriko: commented out to avoid dependency on onig-sys which fails to build on f38
```
Comment (alanconway) [inline lib/vrl/stdlib/Cargo.toml:127]: Can we just delete these if not needed? Any reason we would want to bring them back?

File: src/tls/mod.rs [hunk 2]
```diff
@@ -168,6 +168,32 @@ pub(crate) fn tls_connector_builder(settings: &MaybeTlsSettings) -> Result<SslCo
     let mut builder = SslConnector::builder(SslMethod::tls()).context(TlsBuildConnectorSnafu)?;
     if let Some(settings) = settings.tls() {
         settings.apply_context(&mut builder)?;
+        let mut min_proto_version = SslVersion::TLS1;
```
Comment (alanconway) [inline src/tls/mod.rs:171]: Repetitive, could you extract a re-usable function or method returning min_version + ciphersuites? Pardon my Rust, I don't know what that would look like.

**Review by alanconway — CHANGES_REQUESTED** [review]
Looks good, suggest collecting common logic into one place.

File: src/tls/mod.rs (hunk 2 above)
Comment (syedriko) [reply src/tls/mod.rs]: @alanconway Please take another look.

**Review by alanconway — APPROVED** [review]

**Pull request merged**